### PR TITLE
Add compose example for ChatQnA AMD ROCm deployment

### PR DESCRIPTION
**Pull request by artem-astafev**
## Type of change

- Added compose file for AMD ROCm docker compose deployment.
- Added Readme.md for AMD ROCm docker compose deployment.
- Added dedicated test script for AMD ROCm

## Tests

For functional test go to /GenAIExamples/ChatQnA/tests/ dir and run test_compose_on_rocm.sh script.

### CODE REVIEW

File: ChatQnA/tests/test_compose_on_rocm.sh [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+
+function stop_docker() {
+    cd "$WORKPATH"/docker_compose/amd/gpu/rocm
+    docker compose stop && docker compose rm -f
+}
+
+function main() {
+
+    stop_docker
+#    if [[ "$IMAGE_REPO" == "opea" ]]; then build_docker_images; fi
```
Comment (chensuyue) [inline ChatQnA/tests/test_compose_on_rocm.sh:239]: open this one.

File: ChatQnA/docker_compose/amd/gpu/rocm/compose.yaml [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+services:
+  chatqna-redis-vector-db:
+    image: redis/redis-stack:7.2.0-v9
+    container_name: redis-vector-db
+    ports:
+      - "${CHATQNA_REDIS_VECTOR_PORT}:6379"
+      - "${CHATQNA_REDIS_VECTOR_INSIGHT_PORT}:8001"
+  chatqna-dataprep-redis-service:
+    image: ${REGISTRY:-opea}/dataprep-redis:latest
```
Comment (chensuyue) [inline ChatQnA/docker_compose/amd/gpu/rocm/compose.yaml:12]: ${REGISTRY:-opea}/llm-docsum-tgi:${TAG:-latest}

Comment (artem-astafev) [reply ChatQnA/docker_compose/amd/gpu/rocm/compose.yaml]: Done, tag reverted to ${TAG:-latest}

File: ChatQnA/tests/test_compose_on_rocm.sh (hunk 1 above)
Comment (artem-astafev) [reply ChatQnA/tests/test_compose_on_rocm.sh]: done, build step is enabled

**Comment by chensuyue** [comment]
https://github.com/opea-project/GenAIExamples/actions/runs/11836663420/job/33019777119?pr=1122 Please fix those issue.

File: ChatQnA/docker_compose/amd/gpu/rocm/README.md [hunk 3]
```diff
@@ -0,0 +1,434 @@
+###### Copyright (C) 2024 Advanced Micro Devices, Inc.
```
Comment (chensuyue) [inline ChatQnA/docker_compose/amd/gpu/rocm/README.md:1]: Please remove the copyright from docs.

Comment (chensuyue) [reply ChatQnA/docker_compose/amd/gpu/rocm/README.md:1]: This will block the doc IO build.

**Review by lkk12014402 — APPROVED** [review]

File: ChatQnA/docker_compose/amd/gpu/rocm/README.md [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+
+```yaml
+  chaqna-react-ui-server:
+    image: opea/chatqna-react-ui:latest
+    ...
+    ports:
+      - "80:80"
+```
+
+![project-screenshot](../../../assets/img/chat_ui_init.png)
```
Comment (chensuyue) [inline ChatQnA/docker_compose/amd/gpu/rocm/README.md:426]: https://github.com/opea-project/GenAIExamples/actions/runs/11836663420/job/33019777119?pr=1122 please check the link issue.

File: ChatQnA/docker_compose/amd/gpu/rocm/README.md (hunk 3 above)
Comment (artem-astafev) [reply ChatQnA/docker_compose/amd/gpu/rocm/README.md]: Removed, done

File: ChatQnA/docker_compose/amd/gpu/rocm/README.md (hunk 4 above)
Comment (artem-astafev) [reply ChatQnA/docker_compose/amd/gpu/rocm/README.md:424]: I suppose it related to frontend validation, I've enabled it , but can not find output for path in new run

Comment (chensuyue) [reply ChatQnA/docker_compose/amd/gpu/rocm/README.md:424]: Now, both Doc build CI and hyper link check CI point to the same issue.

Comment (artem-astafev) [reply ChatQnA/docker_compose/amd/gpu/rocm/README.md]: @chensuyue thank you for the info, seems i've find the cause of the issue, as we have deeper dir we need to use one more ../ to get to asset path correctly

**Pull request merged**